hivemq-edge/src/main/java/com/hivemq/configuration/reader/LegacyConfigFileReaderWriter.java

```diff
@@ -0,0 +1,200 @@
+
+    public synchronized @NotNull LEGACY_CONFIG_CLASS readConfigFromXML() {
+        if (configurationFile.file().isEmpty()) {
+            log.error("No configuration file present. Shutting down HiveMQ Edge.");
+            throw new UnrecoverableException(false);
+        }
+        final File configFile = configurationFile.file().get();
+        log.debug("Reading configuration file {}", configFile);
+        final List<ValidationEvent> validationErrors = new ArrayList<>();
+
+        try {
+            final JAXBContext context = createContext();
+            final Unmarshaller unmarshaller = context.createUnmarshaller();
+            //replace environment variable placeholders
+            String configFileContent = Files.readString(configFile.toPath());
+            configFileContent = EnvVarUtil.replaceEnvironmentVariablePlaceholders(configFileContent);
+            final ByteArrayInputStream is =
+                    new ByteArrayInputStream(configFileContent.getBytes(StandardCharsets.UTF_8));
+            final StreamSource streamSource = new StreamSource(is);
+
+            unmarshaller.setEventHandler(e -> {
+                if (e.getSeverity() > ValidationEvent.ERROR) {
+                    validationErrors.add(e);
+                }
+                return true;
+            });
+
+            final JAXBElement<? extends LEGACY_CONFIG_CLASS> result =
+                    unmarshaller.unmarshal(streamSource, legacyConfigClass);
+
+            if (!validationErrors.isEmpty()) {
+                throw new JAXBException("Parsing failed");
+            }
+            return result.getValue();
+        } catch (final JAXBException | IOException e) {
+            final StringBuilder messageBuilder = new StringBuilder();
+
+            if (validationErrors.isEmpty()) {
+                messageBuilder.append("of the following error: ");
+                messageBuilder.append(requireNonNullElse(e.getCause(), e));
+            } else {
+                messageBuilder.append("of the following errors:");
+                for (final ValidationEvent validationError : validationErrors) {
+                    messageBuilder.append(System.lineSeparator()).append(toValidationMessage(validationError));
+                }
+            }
+            log.error("Not able to parse configuration file because {}", messageBuilder);
+            throw new UnrecoverableException(false);
+        } catch (final Exception e) {
+
+            if (e.getCause() instanceof UnrecoverableException) {
+                if (((UnrecoverableException) e.getCause()).isShowException()) {
+                    log.error("An unrecoverable Exception occurred. Exiting HiveMQ", e);
+                    log.debug("Original error message:", e);
+                }
+                System.exit(1);
+            }
+            log.error("Could not read the configuration file {}. Exiting HiveMQ Edge.", configFile.getAbsolutePath());
+            log.debug("Original error message:", e);
```

Comment:
Let's add this to the. error log.
This information will be vital in case we really run int osomething like that.

Reply:
that is the usual HiveMQ pattern. The stack trace should be on debug. Ofc we can break with that